VideoLocker/src/main/java/org/edx/mobile/discussion/DiscussionTextUtils.java

```diff
@@ -137,8 +135,8 @@ public void updateDrawState(TextPaint ds) {
                         context.getResources(), finalStringRes, valuesMap));
                 // If time is not available, then reduce the whitespaces
                 // surrounding it's placeholder in the template to one.
-                if (TextUtils.isEmpty(formattedText) || TextUtils.isEmpty(authorLabel)) {
-                    formattedText = removeDuplicateWhitespace(formattedText);
+                if (TextUtils.isEmpty(formattedTime)) {
+                    formattedText = removeDuplicateWhitespaces(formattedText);
```

Comment:
Using separate text views would probably be overkill, since it's one logical string after all, and we'll have to write a utility method for setting them anyway.

By the way, @aleffert, I was looking over the iOS joining implementation for this, and it seems to be using the `joinInNaturalLayout()` method, which treats it like separate views, and reverses the join in RTL mode. I don't think that is appropriate for this scenario. 😜